pkg/machinepool/machinepool_test.go

```diff
@@ -340,32 +340,265 @@ var _ = Describe("Machinepool and nodepool", func() {
 				Expect(boolean).To(BeFalse())
 			})
 		})
-		It("Test printMachinePools", func() {
+		It("Test printMachinePools with basic functionality", func() {
 			clusterBuilder := cmv1.NewCluster().ID("test").State(cmv1.ClusterStateReady).
 				MachinePools(cmv1.NewMachinePoolList().
-					Items(cmv1.NewMachinePool().ID("np").Replicas(8).Subnets("sn1", "sn2").
-						InstanceType("test instance type").Taints(cmv1.NewTaint().Value("test").
-						Key("taint"))))
+					Items(cmv1.NewMachinePool().ID("mp-1").Replicas(3).
+						InstanceType("m5.large").Subnets("subnet-1", "subnet-2").
+						Taints(cmv1.NewTaint().Value("test-value").Key("test-key"))))
 			cluster, err := clusterBuilder.Build()
 			Expect(err).ToNot(HaveOccurred())
-			out := getMachinePoolsString(cluster.MachinePools().Slice())
-			Expect(err).ToNot(HaveOccurred())
-			Expect(out).To(Equal(fmt.Sprintf("ID\tAUTOSCALING\tREPLICAS\tINSTANCE TYPE\tLABELS\t\tTAINTS\t"+
-				"\tAVAILABILITY ZONES\t\tSUBNETS\t\tSPOT INSTANCES\tDISK SIZE\tSG IDs\n"+
-				"%s\t%s\t%s\t%s\t%s\t\t%s\t\t%s\t\t%s\t\t%s\t%s\t%s\n",
-				cluster.MachinePools().Get(0).ID(),
-				ocmOutput.PrintMachinePoolAutoscaling(cluster.MachinePools().Get(0).Autoscaling()),
-				ocmOutput.PrintMachinePoolReplicas(cluster.MachinePools().Get(0).Autoscaling(),
-					cluster.MachinePools().Get(0).Replicas()),
-				cluster.MachinePools().Get(0).InstanceType(),
-				ocmOutput.PrintLabels(cluster.MachinePools().Get(0).Labels()),
-				ocmOutput.PrintTaints(cluster.MachinePools().Get(0).Taints()),
-				output.PrintStringSlice(cluster.MachinePools().Get(0).AvailabilityZones()),
-				output.PrintStringSlice(cluster.MachinePools().Get(0).Subnets()),
-				ocmOutput.PrintMachinePoolSpot(cluster.MachinePools().Get(0)),
-				ocmOutput.PrintMachinePoolDiskSize(cluster.MachinePools().Get(0)),
-				output.PrintStringSlice(cluster.MachinePools().Get(0).AWS().AdditionalSecurityGroupIds()))))
+
+			r := &rosa.Runtime{}
+			out := getMachinePoolsString(r, cluster.MachinePools().Slice(), false, false, false, false)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that the output contains the expected basic headers that have data
+			Expect(out).To(ContainSubstring("ID"))
+			Expect(out).To(ContainSubstring("AUTOSCALING"))
+			Expect(out).To(ContainSubstring("REPLICAS"))
+			Expect(out).To(ContainSubstring("INSTANCE TYPE"))
+			Expect(out).To(ContainSubstring("TAINTS"))
+			Expect(out).To(ContainSubstring("SUBNETS"))
+			Expect(out).To(ContainSubstring("SPOT INSTANCES"))
+			Expect(out).To(ContainSubstring("DISK SIZE"))
+
+			// Check that the output contains the machine pool data
+			Expect(out).To(ContainSubstring("mp-1"))
+			Expect(out).To(ContainSubstring("m5.large"))
+			Expect(out).To(ContainSubstring("subnet-1, subnet-2"))
+			Expect(out).To(ContainSubstring("test-key=test-value:"))
```

Comment:
Could you explain the reasoning behind getting rid of the more concise `Equal()` check for the output and replacing it entirely?

Reply:
But isn't it more to the dynamic side when I have columns that can or cannot be shown due to being empty  ?

Reply:
It does, will do the rewrite here, thanks for explaining